### DISCUSSION TITLE
`numeric(9, 3)` causing `Trying to set Object.length of type 'number' to '1.000' of type 'string'`

### DISCUSSION THREAD

**Discussion by tukusejssirs** [mikro-orm/mikro-orm#2659] [Q&A] [answered]
Related: #324

---

I need to store duration in seconds with three decimal places. The duration is usually somewhere in between 10-100 minutes, however, it could be several days too, therefore I chose the `numeric(9, 3)` type for the column. For entity property, I kept it as `number`.

This, howewer, causes runtime error. What is the prefered solution to this use case? Change the property type to `BigInt`? Or to `string` and create a custom type to convert between the values?

Thanks in advance!

**Update:** [`BigInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) does not return fractional digits after division, therefore I cannot use it.

**Comment by B4nan** [comment, marked as answer]
numeric type is mapped to string, as you could loose precision otherwise - js numbers are floats and those can't represent all the values of numeric. bigint is the very same thing, just without fractions, not fully representable by js number. you should map those to strings, if you want to map them to numbers, create your own custom type. 

the reason is mainly large numers (larger than `Number.MAX_SAFE_INTEGER`), if you dont need to care about those, it should be safe to use such custom type and map to numbers. but i would still rather go for regular integer (if you need to have milliseconds precision, you could also store milliseconds as ints instead of seconds as decimals)

numeric(9,3) should mean 9 digits in total, 6 + 3 fractions, right? if so, that should fit into regular integer as milliseconds

**Reply by tukusejssirs** [reply]
> numeric(9,3) should mean 9 digits in total, 6 + 3 fractions, right? if so, that should fit into regular integer as milliseconds

You’re right, it didn’t cross my mind to use milliseconds instead of seconds, but yeah, that’ll ease some other stuff too.

Thanks for your suggestion and help! :pray:
